core/src/main/java/com/linecorp/armeria/internal/server/annotation/ResponseConverterFunctionSelector.java

```diff
@@ -0,0 +1,186 @@
+                                                    AnnotatedService.class.getClassLoader()));
+
+    static final List<ResponseConverterFunctionProvider>
+            responseConverterFunctionProviders =
+            ImmutableList.copyOf(ServiceLoader.load(ResponseConverterFunctionProvider.class,
+                                                    AnnotatedService.class.getClassLoader()));
+
+    static {
+        if (!delegatingResponseConverterFunctionProviders.isEmpty()) {
+            logger.debug("Available {}s: {}", DelegatingResponseConverterFunctionProvider.class.getSimpleName(),
+                         delegatingResponseConverterFunctionProviders);
+        }
+        if (!responseConverterFunctionProviders.isEmpty()) {
+            logger.debug("Available {}s: {}", ResponseConverterFunctionProvider.class.getSimpleName(),
+                         responseConverterFunctionProviders);
+        }
+    }
+
+    static ResponseConverterFunction responseConverter(
+            Type returnType, List<ResponseConverterFunction> responseConverters) {
+
+        final List<ResponseConverterFunction> nonDelegatingSpiConverters =
+                responseConverterFunctionProviders.stream().map(
+                        provider -> provider.newResponseConverterFunction(returnType)
+                ).filter(Objects::nonNull).collect(Collectors.toList());
+
+        final ImmutableList<ResponseConverterFunction> backingConverters =
+                ImmutableList
+                        .<ResponseConverterFunction>builder()
+                        .addAll(responseConverters)
+                        .addAll(nonDelegatingSpiConverters)
+                        .addAll(defaultResponseConverters)
+                        .build();
+
+        final ResponseConverterFunction responseConverter = new CompositeResponseConverterFunction(
+                ImmutableList
+                        .<ResponseConverterFunction>builder()
+                        .addAll(backingConverters)
+                        // It is the last converter to try to convert the result object into an HttpResponse
+                        // after aggregating the published object from a Publisher or Stream.
+                        .add(new AggregatedResponseConverterFunction(
+                                new CompositeResponseConverterFunction(backingConverters)))
+                        .build());
+
+        for (final DelegatingResponseConverterFunctionProvider provider : delegatingResponseConverterFunctionProviders) {
+            final ResponseConverterFunction func =
+                    provider.createResponseConverterFunction(returnType, responseConverter);
+            if (func != null) {
+                return func;
+            }
+        }
+
+        return responseConverter;
+    }
+
+    /**
+     * A default {@link ResponseConverterFunction}s.
+     */
+    private static final List<ResponseConverterFunction> defaultResponseConverters =
```

Comment:
Could you put this static field above the static block? e.g. line 63

Reply:
Fixed in 849f6b39